[sushiyade] iP

sushiyade opened the pull request:
# Sunacchi
> “Why not do something with your life instead of sleeping everyday” – My Dad 

An unrelated quote to read at your own [leisure](https://dansilvestre.com/productivity-quotes)

Sunacchi frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 🙇‍♂️ 

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ] Reminders (coming soon


If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

sushiyade committed:
level-0

sushiyade committed:
level-1

sushiyade committed:
Removed spaces

sushiyade committed:
level-2

sushiyade committed:
level-2 new

sushiyade committed:
New level-2

sushiyade committed:
level-3

sushiyade committed:
level-4

sushiyade committed:
level-5

sushiyade committed:
.....

sushiyade committed:
level-6

sushiyade committed:
Merge to master

sushiyade committed:
Add usability

sushiyade committed:
Add save function

sushiyade committed:
Add proper imports

sushiyade committed:
Add date time functionality

sushiyade committed:
Change master for non-fastforward merge

sushiyade committed:
Merge tag 'level-8'

sushiyade committed:
Remove extra code

sushiyade committed:
Change to OOP Structure

sushiyade committed:
Add gradle support

sushiyade committed:
Add JUnit Tests

sushiyade committed:
Add JavaDocs

sushiyade committed:
Fix import ordering

sushiyade committed:
Fix import ordering

sushiyade committed:
Add some of level-9

sushiyade committed:
Finish level-9

sushiyade committed:
Merge coding standard with master branch

sushiyade committed:
Merge level-9 with master branch

sushiyade committed:
Add spacing

sushiyade committed:
Add new branches

sushiyade committed:
Add checkstyle functionality

sushiyade committed:
Add dependencies for JavaFX

Bombbird2001 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+public class Deadline extends Task{
+    private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy");
+    protected LocalDate by;
+    /**
```
Perhaps there could be an empty line before this? Also noticed this in some other places

Bombbird2001 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+    @Override
+    public String toString() {
+        return "[D]" + "[" + this.getStatusIcon() + "]"
+                + super.toString() + " (by: " + by.format(DATE_FORMATTER) + ")";
```
I like that this follows the line break standards, with 8 spaces and break before an operator

Bombbird2001 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,53 @@
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
```
Should this empty line(s) be removed? Noticed in other places as well

Bombbird2001 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+ * Represents a task with a specific deadline.
+ * Inherits from the Task class.
+ */
+public class Deadline extends Task{
```
Should there be a space between Task and { ?

Bombbird2001 replied:
Should there be a space between Task and { ?

Bombbird2001 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,53 @@
+package duke;
+
+import java.io.*;
```
Should the imports be declared explicitly instead of using wildcard? Also noticed in a few other files

Bombbird2001 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,160 @@
+ * Handles the parsing of user input and the corresponding actions in the Duke application.
+ */
+public class Parser {
+    final static String EXIT_PHRASE = "bye";
+    final static String LIST_PHRASE = "list";
+    final static String TODO_PHRASE = "todo";
+    final static String DEADLINE_PHRASE = "deadline";
+    final static String EVENT_PHRASE = "event";
+    final static String DELETE_PHRASE = "delete";
+    final static String SEARCH_PHRASE = "find";
```
I like the consistent naming of variables

Bombbird2001 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,59 @@
+ * Handles the saving and loading of tasks to and from a file in a directory.
+ */
+public class Storage {
+    java.nio.file.Path path = java.nio.file.Paths.get( "./","src", "main", "data", "duke.txt");
```
Perhaps we could replace the fully qualified path with an import statement instead?

Bombbird2001 reviewed (commented):
Looks good, just some points to note as mentioned in the comments

sushiyade committed:
level 10

sushiyade committed:
Merge checkstyle

sushiyade committed:
Follow checkstyle guidelines

nathanielcalimag commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+     * @param i The index of the task to remove.
+     * @return Always returns `null`.
+     */
+    public Task remove(int i) {
+        tasks.remove(i);
+        return null;
+    }
```
Should this be `public void remove` instead of `public Task remove` since the return is always `null`? The return value of this function does not seem to be used.

nathanielcalimag commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,136 @@
+                }
+            }
+            return finalText;
+        } else if (commandWord.equals("mark")) {
```
Should these be defined as constants consistency since the other commands are also stored in constants?

nathanielcalimag commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+     */
+    @Override
+    public boolean add(Task task) {
+        tasks.add(task);
```
What is the purpose of the return value for this `add` method?

nathanielcalimag commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,36 @@
+     * It checks if the formatted deadline matches the expected result.
+     */
+    @Test
+    public void testGetFormattedDeadline_taskWithDeadline_formattedDateTime() {
```
I like the comprehensive test method names which make it clear what the tests cover.

nathanielcalimag commented on src/main/java/duke/gui/MainWindow.java:
```diff
@@ -0,0 +1,53 @@
+        duke = d;
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    @FXML
+    private void handleUserInput() throws DukeException {
```
Should this javadoc comment explain when the `DukeException` will be thrown?

For instance,

```java
    /**
     * ...
     * @throws DukeException If ...
     */
```

sushiyade committed:
Enhance Code Reliability with Assertions

We've added assertions to validate assumptions and preconditions in the code. This improves error detection and prevents unexpected issues.

- In the `TaskList` class, we use assertions to ensure tasks are not null and indices are within bounds during manipulation.
- In the `Main` class, assertions validate directory and file creation, as well as task list loading.
- In the `DialogBox` class, we confirm that UI elements are not null after initialisation.

sushiyade committed:
Merge pull request #1 from sushiyade/A-Assertions

Enhance Code Reliability with Assertions

sushiyade committed:
Refactor Parser class to use Enums and Improve Code Readability

Refactored the Parser class to enhance code readability and maintainability by introducing an enum, `CommandType`, to represent the different command types in Duke.

Introduced the `CommandType` enum, which encapsulates the possible command types such as EXIT, LIST, MARK, UNMARK, DELETE, TODO, DEADLINE, EVENT, SEARCH, and UNKNOWN.
Created the `getCommandType` method to determine the command type based on user input. This method trims and converts the input to lowercase for consistent comparisons.
Modified the `parse` method to use a switch statement for handling each command type individually.
Added handling for the UNKNOWN command type, providing a clear error message when an unknown command is encountered.

These changes make the Parser class more modular, easier to follow, and align with coding quality standards.

sushiyade committed:
Merge branch 'master' into A-CodeQuality

sushiyade committed:
Merge pull request #2 from sushiyade/A-CodeQuality

Refactor Parser class to use Enums and Improve Code Readability

sushiyade committed:
change formatting of datetime in Event

sushiyade committed:
Add fixed duration tasks functionality

sushiyade committed:
Add Duke starting message

sushiyade committed:
Include user guide for Sunacchi

sushiyade committed:
Include user guide for Sunacchi

sushiyade committed:
Fix Jar storage paths

sushiyade committed:
add JavaDoc

sushiyade committed:
Change order of imports in Storage.java

sushiyade committed:
Fix bugs

sushiyade committed:
Fix bugs
